lighthouse-core/gather/drivers/driver.js

```diff
@@ -210,6 +211,42 @@ class Driver {
   }
 
   /**
+   * Checks all serviceworkes and see if any duplications are running
+   * @param {string} pageUrl
+   */
+  checkForMultipleServiceWorkers(pageUrl) {
+    // Get necessary information of serviceWorkers
+    const getRegistrations = new Promise(resolve => {
+      this.once('ServiceWorker.workerRegistrationUpdated', resolve);
+    });
+    const getVersions = this.getServiceWorkerVersions();
+
+    return Promise.all([getRegistrations, getVersions]).then(res => {
+      const registrations = res[0].registrations;
+      const versions = res[1].versions;
+
+      let activeServiceWorkers = [];
+      registrations
+        .filter(reg => !reg.isDeleted)
+        .filter(reg => parseURL(reg.scopeURL).hostname === parseURL(pageUrl).hostname)
```

Comment:
oh, I think @paulirish had something about using just hostname to do some limited redirect handling. One issue is port number...e.g. our smoketest on `localhost:whatever` might alert on this even though one of them has a SW and one doesn't